python/python/lance/dataset.py

```diff
@@ -867,6 +867,7 @@ def create_index(
         replace: bool = False,
         num_partitions: Optional[int] = None,
         ivf_centroids: Optional[Union[np.ndarray, pa.FixedSizeListArray]] = None,
+        pq_codebook: Optional[Union[np.ndarray, pa.FloatArray]] = None,
```

Comment:
Can we standardize ivf_centroids , pq_codebook to take the same data types,

including, `np.ndarray`, `pa.FixedSizeListArray`, and `pa.FixedShapedTensor`? making them different (as exposing the internal details) is confusing.